satellite/orders/endpoint.go

```diff
@@ -199,21 +201,22 @@ func (endpoint *Endpoint) Settlement(stream pb.Orders_SettlementServer) (err err
 				if err != nil {
 					return formatError(err)
 				}
-			} else {
-				return err
+				continue
 			}
-			continue
+			return err
 		}
 		now := time.Now().UTC()
 		intervalStart := time.Date(now.Year(), now.Month(), now.Day(), now.Hour(), 0, 0, 0, now.Location())
 
 		if err := endpoint.DB.UpdateBucketBandwidthSettle(ctx, bucketID, orderLimit.Action, order.Amount, intervalStart); err != nil {
+			// TODO: we should use the serial number in the same transaction we settle the bandwidth? that way we don't need this undo in an error case?
 			if err := endpoint.DB.UnuseSerialNumber(ctx, orderLimit.SerialNumber, orderLimit.StorageNodeId); err != nil {
 				log.Error("unable to unuse serial number", zap.Error(err))
 			}
 			return err
 		}
 
+		// TODO: whoa this should also be in the same transaction
```

Comment:
Wouldn't using transactions make it slower?

The current logic is implemented to try and avoid transactions for performance, but yeah it could be batched or a single transaction.

Reply:
i mean ideally it's all in the same query even, but yes sharing a transaction will drastically improve performance

Reply:
Oh, interesting.